File: app/src/main/java/jp/gcreate/sample/kotlindaggerdatabinding/di/AppComponent.java
package jp.gcreate.sample.kotlindaggerdatabinding.di;

import javax.inject.Singleton;

import dagger.Component;
import jp.gcreate.sample.kotlindaggerdatabinding.MainActivity;

/**
 * Copyright 2016 G-CREATE
 *
 */
@Singleton
@Component(modules = AppModule.class)
public interface AppComponent {
    void inject(MainActivity mainActivity);
}

File: app/src/main/java/jp/gcreate/sample/kotlindaggerdatabinding/di/AppComponent.kt
package jp.gcreate.sample.kotlindaggerdatabinding.di

import dagger.Component
import jp.gcreate.sample.kotlindaggerdatabinding.MainActivity
import javax.inject.Singleton

/**
 * Copyright 2016 G-CREATE
 *
 */
@Singleton
@Component(modules = arrayOf(AppModule::class))
interface AppComponent {
    fun inject(mainActivity: MainActivity)
}